cluster/validation.go

```diff
@@ -13,16 +15,79 @@ func (c *Cluster) ValidateCluster() error {
 	if len(c.EtcdHosts) == 0 {
 		return fmt.Errorf("Cluster must have at least one etcd host")
 	}
+	if len(c.WorkerHosts) == 0 {
+		return fmt.Errorf("Cluster must have at least one worker plane host")
+	}
+
+	// validate hosts options
+	if err := validateHostsOptions(c); err != nil {
+		return err
+	}
+
+	// validate Auth options
+	if err := validateAuthOptions(c); err != nil {
+		return err
+	}
+
+	// validate Network options
+	if err := validateNetworkOptions(c); err != nil {
+		return err
+	}
 
 	// validate services options
-	err := validateServicesOption(c)
-	if err != nil {
+	if err := validateServicesOptions(c); err != nil {
 		return err
 	}
 	return nil
 }
 
-func validateServicesOption(c *Cluster) error {
+func validateAuthOptions(c *Cluster) error {
+	if c.Authentication.Strategy != DefaultAuthStrategy {
+		return fmt.Errorf("Authentication strategy [%s] is not supported", c.Authentication.Strategy)
+	}
+	return nil
+}
+
+func validateNetworkOptions(c *Cluster) error {
+	if c.Network.Plugin != "flannel" && c.Network.Plugin != "calico" && c.Network.Plugin != "canal" {
+		return fmt.Errorf("Network plugin [%s] is not supported", c.Network.Plugin)
+	}
+	return nil
+}
+
+func validateHostsOptions(c *Cluster) error {
+	for i, host := range c.Nodes {
+		if len(host.Address) == 0 {
+			return fmt.Errorf("User for host (%d) is not provided", i+1)
+		}
+		if len(host.User) == 0 {
+			return fmt.Errorf("User for host (%d) is not provided", i+1)
+		}
+		if len(host.Role) == 0 {
+			return fmt.Errorf("Role for host (%d) is not provided", i+1)
+		}
+		for _, role := range host.Role {
+			if role != services.ETCDRole && role != services.ControlRole && role != services.WorkerRole {
+				return fmt.Errorf("Role [%s] for host (%d) is not recognized", role, i+1)
+			}
+		}
+		bothRoles := make(map[string]bool)
```

Comment:
any reason why we use a map here? there are only 2 values, workerRole and controlRole, may be can use bool instead to simplify the outlook?